src/drivers/actuators/vertiq_io/vertiq_io.hpp

```diff
@@ -0,0 +1,217 @@
+#include "iq-module-communication-cpp/inc/pulsing_rectangular_input_parser_client.hpp"
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+enum DISARM_BEHAVIORS {USER_MIXER_VALUE, TRIGGER_MOTOR_DISARM, COAST_MOTOR, SEND_PREDEFINED_THROTTLE};
+enum ARM_BEHAVIORS {USE_MOTOR_ARMING, FORCE_ARMING};
+
+class VertiqIo : public ModuleBase<VertiqIo>, public OutputModuleInterface
+{
+
+public:
+	/**
+	* @brief Create a new VertiqIo object
+	*/
+	VertiqIo();
+
+	/**
+	* @brief destruct a VertiqIo object
+	*/
+	~VertiqIo() override;
+
+	/**
+	* @brief initialize the VertiqIo object. This will be called by the task_spawn function. Makes sure that the thread gets scheduled.
+	*/
+	bool init();
+
+	/** @see ModuleBase */
+	static int task_spawn(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	int print_status() override;
+
+	/** @see ModuleBase::run() */ //I do not think this actually comes from ModuleBase. it should come from scheduled work item
+	void Run() override;
+
+	/** @see OutputModuleInterface */
+	bool updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS],
+			   unsigned num_outputs, unsigned num_control_groups_updated) override;
+
+	/**
+	* @brief Used to package and transmit controls via IQUART
+	* @param outputs The output throttles calculated by the mixer
+	*/
+	void OutputControls(uint16_t outputs[MAX_ACTUATORS]);
+
+private:
+
+	/**
+	 * Check for parameter changes and update them if needed.
+	 */
+	void parameters_update();
+
+	perf_counter_t	_loop_perf{perf_alloc(PC_ELAPSED, MODULE_NAME": cycle")};
+	perf_counter_t	_loop_interval_perf{perf_alloc(PC_INTERVAL, MODULE_NAME": output update interval")};
+
+	static const uint8_t MAX_SUPPORTABLE_IFCI_CVS = 16;
+
+	static px4::atomic_bool _request_telemetry_init; //Determines whether or not we should initialize or re-initialize the serial connection
+
+	MixingOutput _mixing_output{"VERTIQ_IO", MAX_SUPPORTABLE_IFCI_CVS, *this, MixingOutput::SchedulingPolicy::Auto, false, false};
+
+	static char _telemetry_device[20]; //The name of the device we're connecting to. this will be something like /dev/ttyS3
+
+	VertiqSerialInterface _serial_interface; //We need a serial handler in order to talk over the serial port
+	VertiqClientManager _client_manager; //We need someone who can manage our clients
+	VertiqTelemetryManager _telem_manager; //We need a telemetry handler
+	VertiqConfigurationHandler _configuration_handler;
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq client information
+
+/**
+In order to communicate with connected Vertiq modules with the most possible flexibility, we have introduced two types of Clients (see comment above about
+Vertiq clients and entries) to be used within PX4. We have designated "Operational Clients" as those whose object IDs (a.k.a target module IDs) are constant.
```

Comment:
Is this comment quite accurate still? I suppose the telemetry IFCI one sort of breaks the distinction since it changes. Now that we don't have the different AddOperational vs. AddConfiguration functions and instead just have them split out into classes, may not need this comment anymore, or at least it should be updated probably